### PR TITLE
Add tests to Utils

### PR DESCRIPTION
**Pull request by QubitPi**
https://github.com/yahoo/fili/issues/700

### CODE REVIEW

File: fili-core/src/test/groovy/com/yahoo/bard/webservice/util/UtilsSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        and: "getting by HashSet returns both HashSet and LinkedHashSet instances"
+        Utils.getSubsetByType(set, HashSet) == set as LinkedHashSet
+    }
+
+    def "File path without parent dir generates the parent"() {
+        given: "a file path without a parent directory"
+        String path = TEMP_DIR + "data.txt"
+
+        // make sure parent directory does not exist yet
+        assert ! new File(TEMP_DIR).exists()
```
Comment (cdeszaq) [inline fili-core/src/test/groovy/com/yahoo/bard/webservice/util/UtilsSpec.groovy]: (minor stylistic) You can make this an `expect` block and and turn the comment into a block comment so you don't need the `assert`

File: fili-core/src/test/groovy/com/yahoo/bard/webservice/util/UtilsSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        String subDir = TEMP_DIR + "subDir/"
+        String subFile = TEMP_DIR + "subDir/data.txt"
+
+        and: "create the directory contents"
+        Files.createFile(Paths.get(file1))
+        Files.createFile(Paths.get(file2))
+        Files.createDirectory(Paths.get(subDir))
+        Files.createFile(Paths.get(subFile))
+
+        // make sure all files and sub-dir exist
```
Comment (cdeszaq) [inline fili-core/src/test/groovy/com/yahoo/bard/webservice/util/UtilsSpec.groovy]: ditto using an `expect` block after a `given` or `setup` block to assert preconditions for a test.

File: fili-core/src/test/groovy/com/yahoo/bard/webservice/util/UtilsSpec.groovy (hunk 1 above)
Comment (QubitPi) [reply fili-core/src/test/groovy/com/yahoo/bard/webservice/util/UtilsSpec.groovy]: I like that. Thanks!

**Comment by QubitPi** [comment]
Thank you, @cdeszaq , for the review and approval!

**Pull request merged**